analyzers/src/SonarAnalyzer.Common/Helpers/ProjectConfigReader.cs

```diff
@@ -32,54 +30,40 @@ namespace SonarAnalyzer.Helpers
     /// </summary>
     internal class ProjectConfigReader
     {
-        private const string SonarProjectConfigFileName = "SonarProjectConfig.xml";
-
-        private readonly Lazy<ProjectConfig> projectConfig;
+        private readonly ProjectConfig projectConfig;
         private readonly Lazy<ProjectType> projectType;
+        private readonly Lazy<FilesToAnalyzeProvider> filesToAnalyze;
 
-        public string AnalysisConfigPath => projectConfig.Value.AnalysisConfigPath;
-        public string FilesToAnalyzePath => projectConfig.Value.FilesToAnalyzePath;
-        public string OutPath => projectConfig.Value.OutPath;
-        public string ProjectPath => projectConfig.Value.ProjectPath;
-        public string TargetFramework => projectConfig.Value.TargetFramework;
+        public string AnalysisConfigPath => projectConfig.AnalysisConfigPath;
+        public string FilesToAnalyzePath => projectConfig.FilesToAnalyzePath;
+        public string OutPath => projectConfig.OutPath;
+        public string ProjectPath => projectConfig.ProjectPath;
+        public string TargetFramework => projectConfig.TargetFramework;
         public ProjectType ProjectType => projectType.Value;
+        public FilesToAnalyzeProvider FilesToAnalyze => filesToAnalyze.Value;
 
-        public ProjectConfigReader(AnalyzerOptions options)
+        public ProjectConfigReader(SourceText sonarProjectConfig, string logFileName)
```

Comment:
I would say the logFileName is always SonarProjectConfigFileName, so maybe it is more appropriate to name it like that. This way it is a bit confusing which file it is.

Reply:
It will always have that value, but `ProjectConfigReader` doesn't know that anymore for sure. It's coming from outside, because it can't be read from `SourceText`.